crates/bevy_ecs/src/world/mod.rs

```diff
@@ -3137,7 +3138,14 @@ impl World {
     /// This can easily cause systems expecting certain resources to immediately start panicking.
     /// Use with caution.
     pub fn clear_resources(&mut self) {
-        self.storages.resources.clear();
+        let pairs: Vec<(ComponentId, Entity)> = self
+            .resource_entities()
+            .iter()
+            .map(|(id, entity)| (*id, *entity))
+            .collect();
+        for (component_id, entity) in pairs {
+            self.entity_mut(entity).remove_by_id(component_id);
+        }
         self.storages.non_send_resources.clear();
```

Comment:
As stated elsewhere, coupling NonSend with Resources is increasingly confusing here. We should break out "non send" into its own set of apis.